src/value/de.rs

```diff
@@ -41,22 +41,19 @@ impl<'de> Deserialize<'de> for Value {
             fn visit_i64<E>(self, i: i64) -> Result<Value, E>
                 where E: SError,
             {
-                Ok(Value::I64(i))
+                Ok(Value::Number(Number::from(i)))
             }
 
             fn visit_u64<E>(self, u: u64) -> Result<Value, E>
                 where E: SError,
             {
-                match NumCast::from(u) {
-                    Some(i) => Ok(Value::I64(i)),
-                    None => Ok(Value::String(u.to_string())),
-                }
+                Ok(Value::Number(Number::from(u)))
             }
 
             fn visit_f64<E>(self, f: f64) -> Result<Value, E>
                 where E: SError,
             {
-                Ok(Value::F64(f))
+                Ok(Value::Number(Number::from_f64(f).expect("wrapping an existing float")))
```

Comment:
This does not work for infinite values! cf. http://yaml.org/type/float.html for valid syntax. I locally added this example to the `as_f64` docs of value:

```rust
/// ```rust
/// # use std::f64;
/// # fn yaml(i: &str) -> serde_yaml::Value { serde_yaml::from_str(i).unwrap() }
/// assert_eq!(yaml("inf").as_f64(), Some(f64::INFINITY));
/// assert_eq!(yaml("-inf").as_f64(), Some(f64::NEG_INFINITY));
/// assert!(yaml("NaN").as_f64().unwrap().is_nan());
/// ```
```